### PR TITLE
feat(SentimentAnalyzer): Improve performance

### PR DESCRIPTION
**Pull request by kamranayub**
## Changes

- Use singleton pattern ([4th version](https://csharpindepth.com/articles/singleton))
- Add benchmarking test for comparison using [imdb-data](https://github.com/nas5w/imdb-data)
- Update to ML.NET@1.5.5
- Update README a bit

## Results

- **Before:** I didn't even let it finish but estimate was 100 minutes 😅
- **After:** ~16 seconds

Specs: core i7-3770k @ 3.5ghz, 32gb ram

### CODE REVIEW

File: Cognitive-Library/SentimentAnalyzer/Sentiments.cs [hunk 1]
```diff
@@ -9,26 +9,47 @@
...
 
-            var mlContext = new MLContext(seed: 1);
-            Sentiment statement = new Sentiment { Col0 = text };
-            ITransformer trainedModel = mlContext.Model.Load(stream, out var modelInputSchema);
+        private Sentiments()
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            using (var stream = assembly.GetManifestResourceStream("SentimentAnalyzer.MLModels.SentimentModel.zip"))
+            {
+                var mlContext = new MLContext();
```
Comment (kamranayub) [inline Cognitive-Library/SentimentAnalyzer/Sentiments.cs:34]: According to docs for `MLContext`, `seed` is not used on an already-trained model so I left it out.

File: Cognitive-Library/SentimentAnalyzer/Sentiments.cs [hunk 2]
```diff
@@ -9,26 +9,47 @@
...
 
-            stream.Close();
+                stream.Close();
+            }
+        }
 
-            //Score
-            return predEngine.Predict(statement);
+        /// <summary>
+        /// Perform a single prediction of the given text
```
Comment (kamranayub) [inline Cognitive-Library/SentimentAnalyzer/Sentiments.cs:45]: A future enhancement could be to accept `string[]` and do a batch prediction with `IDataView` as well.

**Comment by kamranayub** [comment]
Thanks for a simple library @arafattehsin! I was wondering what data model this is trained on? There is a list here of ones that could be used for other options:

https://kavita-ganesan.com/user-review-datasets/

**Review by arafattehsin — APPROVED** [review]
Thanks @kamranayub for such a great improvement. I definitely missed this one. I think I used Yelp Dataset to train this along with some other (which I can't recall) but it was picked up from Kaggle. 

From your pointer, I just thought that we could also bring all of the reviews in one library but from different models. I have a strategy for that in my mind which I will shortly write in an issue. Please feel free to chip in your thoughts.

File: Cognitive-Library/SentimentAnalyzer/Sentiments.cs (hunk 1 above)
Comment (arafattehsin) [reply Cognitive-Library/SentimentAnalyzer/Sentiments.cs:34]: Makes sense.

File: Cognitive-Library/SentimentAnalyzer/Sentiments.cs (hunk 2 above)
Comment (arafattehsin) [reply Cognitive-Library/SentimentAnalyzer/Sentiments.cs:45]: Yes, sounds great. We can do this for future.

File: Cognitive-Library/SentimentAnalyzer/Sentiments.cs [hunk 3]
```diff
@@ -9,26 +9,47 @@
...
 {
-    public class Sentiments
+    /// <summary>
+    /// Thread-safe singleton to manage initialization of ML training model
+    /// to improve runtime performance.
+    /// </summary>
+    public sealed class Sentiments
     {
-        public static SentimentPrediction Predict(string text)
+        private static readonly Sentiments instance = new Sentiments();
```
Comment (ayende) [inline Cognitive-Library/SentimentAnalyzer/Sentiments.cs:18]: This needs to be `[ThreadStatic]` and initialized once per thread

Comment (kamranayub) [reply Cognitive-Library/SentimentAnalyzer/Sentiments.cs:18]: Thanks, addressed in https://github.com/arafattehsin/CognitiveRocket/pull/5

**Pull request merged**